web/src/features/moreCast2/components/TabbedDataGrid.tsx

```diff
@@ -350,22 +344,42 @@ const TabbedDataGrid = ({ morecast2Rows, fromTo, setFromTo, fetchWeatherIndeterm
 
   /********** End useEffects for managing visibility of column groups *************/
 
+  const updateColumnHelper = (editedRows: MoreCast2Row[]) => {
+    storedDraftForecast.updateStoredDraftForecasts(editedRows, getDateTimeNowPST())
+    // Create a copy of all Morecast2ForecastRows
+    let newRows = cloneDeep(allRows)
+    newRows = newRows.map(newRow => editedRows.find(row => row.id === newRow.id) || newRow)
+
+    const rowsForSimulation = filterAllVisibleRowsForSimulation(editedRows) ?? []
+
+    if (rowsForSimulation.length > 0) {
+      const filteredRowsWithIndices = simulateFireWeatherIndices(rowsForSimulation)
+      // Merge the copy of allRows with rows that were updated with simulated indices
+      newRows = newRows.map(newRow => filteredRowsWithIndices.find(row => row.id === newRow.id) || newRow)
+    }
+
+    setAllRows(newRows)
+  }
+
   // Persistence forecasting. Get the most recent actual and persist it through the rest of the
   // days in this forecast period.
   const updateColumnFromLastActual = (forecastField: keyof MoreCast2Row, actualField: keyof MoreCast2Row) => {
-    const newRows = [...visibleRows]
+    const newVisibleRows = cloneDeep(visibleRows)
     // Group our visible rows by station code and work on each group sepearately
-    const groupedByStationCode = groupBy(newRows, 'stationCode')
+    const groupedByStationCode = groupBy(newVisibleRows, 'stationCode')
     for (const values of Object.values(groupedByStationCode)) {
       // Get rows with actuals that have non-NaN values
       const rowsWithActuals: MoreCast2Row[] = values.filter(value => !isNaN(value[actualField] as number))
       // Filter for the row with the most recent forDate as this contains our most recent actual
-      const mostRecentRow = rowsWithActuals.reduce((a, b) => {
-        return a.forDate > b.forDate ? a : b
-      })
+      let mostRecentRow: MoreCast2Row | undefined = undefined
+      if (rowsWithActuals.length > 0) {
+        mostRecentRow = rowsWithActuals.reduce((a, b) => {
```

Comment:
Sonarcloud is flagging this because we have to default start value to reduce to here -- which makes me wonder if we should use reduce for sorting (I think there's some sentry errors around this code as well). Could defer to a follow up PR too.